### DISCUSSION TITLE
Improvement for action masking ?

### DISCUSSION THREAD

**Discussion by slinlee** [PathmindAI/nativerl#361] [Q&A]
@SaharEs, you mentioned this in issue #350. Do you think this would make action masks easier to use and implement for users?
> Also, if masking array was a jagged array instead one long array, we could do the check for each decision separately.
So for example here we can check that the first element of masking array is of size 4 and the second one is of size 3.

**Comment by SaharEs** [comment]
Yes, I think so. Having a separate array for each decision would make it easier to implement, debug, and validate. This becomes a pain specially in models with many decisions like Felipe's model.

Maybe users can even decide to not have masking for some of the decisions! I think this would be useful too.
